code/model.py

```diff
@@ -20,26 +21,24 @@ def train(self, emails, w2v):
         It should print the objective function value as well"""
         pass
 
-    @abstractmethod
-    def save(self, filename):
-        """Save the model and any supporting data structures required to recreate the state
-        This might be very useful for partial training and retraining across runs on the cluster
-        Might need to even save seeds and optimizer state"""
-        pass
-
-    @abstractmethod
-    def load(self, filename):
-        """Load the model and any supporting data structures to recreate the state
-        This might be very useful for partial training and retraining across runs on the cluster
-        Might need to even restore seeds and optimizer state"""
-        pass
-
     @abstractmethod
     def get_repr(self, identifier):
         """Given an identifier(?), get the sender or recipient's vector representation
         TODO: Consider refactoring based on requirements"""
         pass
 
+    def save(self, filename):
+        """
+        Saves the model weights to file so that they can be later loaded for inference purposes
+        """
+        torch.save(self.state_dict(), constants.MODEL_DIR + filename)
```

Comment:
How about a different convention to save all embeddings, models etc?
1) Create a directory in the models folder based on run id and perhaps run time params
2) All filenames won't change much and only the directory is changed
3) Code references won't have to be changed at multiple places because constants.MODEL_DIR will point to the specific model for the run.

Reply:
In fact, you're already doing this for important embeddings.
I think doing something similar here will be better especially incase we by mistake reuse a run id
We are less likely to reuse the same hyperparameters

Reply:
And even if we don't give specific run ids, can we have our internal run id to be a concatenation of the user provided run id and the hyper parameters?